api/src/main/java/org/pmiops/workbench/interceptors/TracingInterceptor.java

```diff
@@ -24,72 +18,46 @@
 import org.springframework.web.method.HandlerMethod;
 import org.springframework.web.servlet.AsyncHandlerInterceptor;
 
-// Interceptor to create a trace of the lifecycle of api calls.
 @Service
 public class TracingInterceptor implements AsyncHandlerInterceptor {
-  private static final Tracer tracer = Tracing.getTracer();
-  private static final Logger log = Logger.getLogger(TracingInterceptor.class.getName());
+  private static final Tracer tracer = GlobalOpenTelemetry.getTracer("org.pmiops.workbench");
 
   private final Provider<WorkbenchConfig> workbenchConfigProvider;
 
   @Autowired
   public TracingInterceptor(Provider<WorkbenchConfig> workbenchConfigProvider) {
     this.workbenchConfigProvider = workbenchConfigProvider;
-    try {
-      StackdriverTraceExporter.createAndRegister(StackdriverTraceConfiguration.builder().build());
-    } catch (IOException e) {
-      log.log(Level.WARNING, "Failed to setup tracing", e);
-    }
   }
 
-  /**
-   * @param handler The Swagger-generated ApiController. It contains our handler as a private
-   *     delegate.
-   */
   @Override
   public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler)
       throws Exception {
-    // OPTIONS methods requests don't need traces.
     if (request.getMethod().equals(HttpMethods.OPTIONS)) {
       return true;
     }
-    // All of our Swagger-generated handlers extend the HandlerMethod class, but we'll check this
-    // before casting to avoid runtime exceptions.
     if (!(handler instanceof HandlerMethod)) {
       return true;
     }
 
-    // Create a new scoped span, which will set the span context for any other traces created
-    // from within this request-handling thread.
-    //
-    // This span will end up in Stackdriver with the name format like "Recv.[methodName]" where
-    // methodName is the method name defined in our Swagger YAML.
-    // Example: Recv.syncBillingProjectStatus
     HandlerMethod handlerMethod = (HandlerMethod) handler;
     SpanBuilder requestSpanBuilder =
-        tracer.spanBuilder(handlerMethod.getMethod().getName()).setSpanKind(Span.Kind.SERVER);
+        tracer.spanBuilder(handlerMethod.getMethod().getName()).setSpanKind(SpanKind.SERVER);
     if (workbenchConfigProvider.get().server.traceAllRequests) {
-      requestSpanBuilder.setSampler(Samplers.alwaysSample());
+      requestSpanBuilder.setSpanKind(SpanKind.SERVER);
```

Comment:
is there a similar setSampler method in opentelemtry? or setting the span kind will sample all requests?